### PR TITLE
PE-6517: feat(snapshots)

### PR DESCRIPTION
**Pull request by thiagocarvalhodev**
- validate snapshot existence before syncing from snapshots

--- Releases ---
Android release: https://appdistribution.firebase.google.com/testerapps/1:305132849030:android:6cf0cd5ec064fad3ffce07/releases/11ohs6433ophg

### CODE REVIEW

File: lib/sync/data/snapshot_validation_service.dart [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  Future<List<SnapshotItem>> validateSnapshotItems(
+    List<SnapshotItem> snapshotItems,
+  ) async {
+    List<SnapshotItem> snapshotsVerified = [];
+
+    final futures = snapshotItems.map((snapshotItem) async {
+      final appConfig = _configService.config;
+
+      try {
+        final headers = {'Accept': '*/*'};
```
Comment (karlprieb) [inline lib/sync/data/snapshot_validation_service.dart:22]: Why don't you use the correct content type here?

**Review by karlprieb — COMMENTED** [review]
What will be done with the snapshots that failed?
For what I can see we just ignore them, right?

**Comment by thiagocarvalhodev** [comment]
> What will be done with the snapshots that failed?
For what I can see we just ignore them, right?

Exactly.

File: lib/sync/data/snapshot_validation_service.dart [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  Future<List<SnapshotItem>> validateSnapshotItems(
+    List<SnapshotItem> snapshotItems,
+  ) async {
+    List<SnapshotItem> snapshotsVerified = [];
+
+    final futures = snapshotItems.map((snapshotItem) async {
+      final appConfig = _configService.config;
+
+      try {
+        final headers = {'Accept': 'content-length'};
```
Comment (karlprieb) [inline lib/sync/data/snapshot_validation_service.dart:22]: `Accept` headers are used to tell the server which content type you're willing to accept, not to signalize the headers you want.

File: lib/sync/data/snapshot_validation_service.dart [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+            final headers = {
+              'Range': 'bytes=${lenght - 8}-$lenght',
+            };
+
+            final validationRequest = await http.get(
+              Uri.parse(
+                  '${appConfig.defaultArweaveGatewayUrl}${snapshotItem.txId}'),
+              headers: headers,
+            );
```
Comment (karlprieb) [inline lib/sync/data/snapshot_validation_service.dart:42]: Why this get call is needed? Isn't the head call enough?

**Pull request merged**